doc/sphinx/addendum/type-classes.rst

```diff
@@ -274,23 +274,41 @@ This declares singleton classes for reflexive and transitive relations,
 (see the :ref:`singleton class <singleton-class>` variant for an
 explanation). These may be used as parts of other classes:
 
+.. coqtop:: none
+
+   Set Warnings "-future-coercion-class-field".
+
 .. coqtop:: all
 
    Class PreOrder (A : Type) (R : relation A) :=
      { PreOrder_Reflexive :> Reflexive A R ;
        PreOrder_Transitive :> Transitive A R }.
 
+.. coqtop:: none
+
+   Set Warnings "+future-coercion-class-field".
+
 The syntax ``:>`` indicates that each ``PreOrder`` can be seen as a
 ``Reflexive`` relation. So each time a reflexive relation is needed, a
 preorder can be used instead. This is very similar to the coercion
 mechanism of ``Structure`` declarations. The implementation simply
 declares each projection as an instance.
 
+.. warn:: A coercion will be introduced in future versions when using ':>' in 'Class' declarations. Use '#[global] Existing Instance field.' instead if you don't want the coercion.
+
+   .. deprecated:: 8.16
+
+   In future versions, in addition to the instance, :g:`:>` should also
+   declare a :ref:`coercion<coercions>`, like
+   :ref:`for any other record field<coercions-classes-as-records>`.
+   Use :cmd:`Existing Instance` or :cmd:`Coercion` separately rather
+   than :g:`:>` if you don't want both in the future (this will also
+   avoid this warning).
+
 .. warn:: Ignored instance declaration for “@ident”: “@term” is not a class
 
    Using this ``:>`` syntax with a right-hand-side that is not itself a Class
-   has no effect (apart from emitting this warning). In particular, is does not
-   declare a coercion.
+   has no effect (apart from emitting this warning).
```

Comment:
If you're willing to change this:
The description of :> just above here should be under the Record command, which is where the `:>` syntax is introduced.  Probably should be like this just above the ```In :n:`@field_spec`:``` line.  Indentation matters.

```
   :n:`:>` in :n:`@of_type`
     Explanation...
```

The warning should appear with the other messages under Record.

Reply:
> The description of :> just above here should be under the Record command, [...]

Good point, done.

> The warning should appear with the other messages under Record.

I'm reluctant to do that, this warning is very specific to the Class command, if put under a command, it should rather be the Class than the Record command.

Reply:
Yeah, Class command makes more sense.  Don't take my suggestions blindly :-).